### PR TITLE
KT-12743 Add Intention to convert nullable var to non-nullable lateinit

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-12743

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/ConvertNullablePropertyToLateinitIntention.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.kotlin.types.typeUtil.makeNotNullable
+
+class ConvertNullablePropertyToLateinitIntention : SelfTargetingIntention<KtProperty>(
+    KtProperty::class.java, "Convert to lateinit var"
+) {
+    override fun isApplicableTo(element: KtProperty, caretOffset: Int): Boolean {
+        return !element.hasModifier(KtTokens.LATEINIT_KEYWORD)
```
[the remark this answers is not in the document]
Comment (Bluexin) [reply idea/src/org/jetbrains/kotlin/idea/intentions/ConvertNullablePropertyToLateinitIntention.kt]: > 
> 
> Please make sure this cannot be applied to local variables

Actually, you can have lateinit local variables as long as they're not of primitive types and it'd be weird not to have the intention in this case.

[the remark this answers is not in the document]
Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/intentions/ConvertNullablePropertyToLateinitIntention.kt]: > > Please make sure this cannot be applied to local variables
> 
> Actually, you can have lateinit local variables as long as they're not of primitive types and it'd be weird not to have the intention in this case.

Thanks, it's true from version 1.2

**Comment by mglukhikh** [comment]
I made a mistake about some `lateinit` limitations. Will fix myself.

**Comment by mglukhikh** [comment]
Merged 👍

**Pull request closed without merging**